File: RedSheet.DbEntities/Models/OurTeamMemberBehaviour.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("OurTeamMemberBehaviours", Schema = "dbo")]
	[Serializable]
	public partial class OurTeamMemberBehaviour  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int OurTeamMemberBehaviourId { get; set; } 
	///<summary>
    ///{{Behaviour}}
    ///</summary> 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public Behaviour AgreeableId { get; set; } 
	///<summary>
    ///{{AssertivenessBehaviour}}
    ///</summary> 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public AssertivenessBehaviour AssertivenessId { get; set; } 
	///<summary>
    ///{{ConflictStyleBehviour}}
    ///</summary> 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public ConflictStyleBehviour ConflictStyleId { get; set; } 
	///<summary>
    ///{{Behaviour}}
    ///</summary> 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public Behaviour ConsciousnessId { get; set; } 
	///<summary>
    ///{{Behaviour}}
    ///</summary> 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public Behaviour EmotionalCompetenceId { get; set; } 
	///<summary>
    ///{{Behaviour}}
    ///</summary> 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public Behaviour OpenMindedId { get; set; } 
 
	[RelationshipTableAttribue("OurTeamMembers","dbo")]
	//Relationships
	public int OurTeamMemberId { get; set; } 
 
	[ForeignKey("OurTeamMemberId")]
	public OurTeamMember OurTeamMember { get; set; } 
	///<summary>
    ///{{Behaviour}}
    ///</summary> 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public Behaviour OutingId { get; set; } 
	///<summary>
    ///{{Behaviour}}
    ///</summary> 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public Behaviour PersonalCalmId { get; set; } 
	///<summary>
    ///{{Behaviour}}
    ///</summary> 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public Behaviour SolutionFocusedId { get; set; } 
	///<summary>
    ///{{Behaviour}}
    ///</summary> 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public Behaviour WIllToWinId { get; set; } 

	public OurTeamMemberBehaviour()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/our-team-member-behaviour.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { OurTeamMember,  } from './'
export class OurTeamMemberBehaviour {
    constructor(ourTeamMemberBehaviour?: OurTeamMemberBehaviour )  {
        let properties = [ "ourTeamMemberBehaviourId", "agreeableId", "assertivenessId", "conflictStyleId", "consciousnessId", "emotionalCompetenceId", "openMindedId", "ourTeamMemberId", "outingId", "personalCalmId", "solutionFocusedId", "wIllToWinId",];
        for (let property of properties)
            if (ourTeamMemberBehaviour && ourTeamMemberBehaviour[property])
                this[property] = ourTeamMemberBehaviour[property];
    }
 
	ourTeamMemberBehaviourId : number =   0 ;
 
    @range(0,2147483647)
	agreeableId : number =   undefined;
 
    @range(0,2147483647)
	assertivenessId : number =   undefined;
 
    @range(0,2147483647)
	conflictStyleId : number =   undefined;
 
    @range(0,2147483647)
	consciousnessId : number =   undefined;
 
    @range(0,2147483647)
	emotionalCompetenceId : number =   undefined;
 
    @range(0,2147483647)
	openMindedId : number =   undefined;
 
    @range(0,2147483647)
	ourTeamMemberId : number =   undefined;
	ourTeamMember : OurTeamMember  ;
 
    @range(0,2147483647)
	outingId : number =   undefined;
 
    @range(0,2147483647)
	personalCalmId : number =   undefined;
 
    @range(0,2147483647)
	solutionFocusedId : number =   undefined;
 
    @range(0,2147483647)
	wIllToWinId : number =   undefined;


}
